src/main/java/picard/fingerprint/CrosscheckFingerprints.java

```diff
@@ -162,15 +187,21 @@ protected int doWork() {
         extensions.addAll(Arrays.asList(IOUtil.VCF_EXTENSIONS));
 
         final List<File> unrolledFiles = IOUtil.unrollFiles(INPUT, extensions.toArray(new String[extensions.size()]));
+        final List<File> unrolledFiles2 = IOUtil.unrollFiles(INPUT2, extensions.toArray(new String[extensions.size()]));
 
         final Map<FingerprintIdDetails, Fingerprint> fpMap = checker.fingerprintFiles(unrolledFiles, NUM_THREADS, 1, TimeUnit.DAYS);
+        final Map<FingerprintIdDetails, Fingerprint> fpMap2 = checker.fingerprintFiles(unrolledFiles2, NUM_THREADS, 1, TimeUnit.DAYS);
 
         log.info("Finished generating fingerprints from files, moving on to cross-checking.");
 
         final List<CrosscheckMetric> metrics = new ArrayList<>();
         final int numUnexpected;
 
-        numUnexpected = crossCheckGrouped(fpMap, metrics, getFingerprintIdDetailsStringFunction(CROSSCHECK_BY), CROSSCHECK_BY);
+        if (INPUT2.isEmpty()) {
+            numUnexpected = crossCheckGrouped(fpMap, metrics, getFingerprintIdDetailsStringFunction(CROSSCHECK_BY), CROSSCHECK_BY);
+        } else {
```

Comment:
The loading of unrolledFiles2, and fpMap2 only needs to be done inside this block.

Reply:
yes...but I thought it better to have them by each other given that they are basically the same command twice....but....you're the reviewer!

Reply:
done.

Reply:
I reversed this, since I'd rather the program fail fast, and not after reading and parsing all the files in INPUT.